lib/repository.py

```diff
@@ -81,7 +82,39 @@ def flush(self):
     if graphical:
         splash.Hide()
         sys.stdout = sys.__stdout__
-    print "The retriever is up-to-date"
+    print "\nThe retriever is up-to-date"
+
+
+def update_progressbar(progress):
+    """Shows progressbar given progress in float
+
+    accepts a a number between 0 and 1 and it's converted to a float.
+    a number under 0 represents a 'halt'.
+    a number at 1 or greater represents 100%
+    """
+    # Modify this to change the default length of the progress bar
+    bar_length = 20
+    # Try to set the bar_length according to the console size
+    try:
+        rows, columns = os.popen('stty size', 'r').read().split()
+        bar_length = int(columns) - 35
+        if(not (bar_length > 1)):
+            bar_length = 20
+    except:
+        pass
+    status = ""
```

Comment:
if we are creating `status`, then I guess we may include it in line   ` text = "\rDownload Progress:....` 
or we can remove the creation of `status`